pkg/controller/core.oam.dev/v1alpha2/core/traits/traitdefinition/testdata/example.go

```diff
@@ -0,0 +1,79 @@
+/*
+ Copyright 2021. The KubeVela Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package testdata
+
+import (
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+// FooSpec describe Foo Status
+type FooSpec struct {
+	Key string `json:"key,omitempty"`
+}
+
+// FooStatus describe Foo Status
+type FooStatus struct {
+	Key     string `json:"key,omitempty"`
+	AppHash string `json:"app-hash,omitempty"`
+}
+
+// Foo describe foo
+type Foo struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Spec FooSpec `json:"spec,omitempty"`
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Status FooStatus `json:"status,omitempty"`
+}
+
+// FooList contains a list of Foo
+type FooList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Foo `json:"items"`
+}
+
+// Package type metadata.
+const (
+	Group   = "example.com"
+	Version = "v1"
+)
+
+var (
+	// SchemeGroupVersion is group version used to register these objects
+	SchemeGroupVersion = schema.GroupVersion{Group: Group, Version: Version}
+
+	// SchemeBuilder is used to add go types to the GroupVersionKind scheme
+	SchemeBuilder = &scheme.Builder{GroupVersion: SchemeGroupVersion}
+)
+
+// FooKind type metadata
+var (
+	FooKind             = reflect.TypeOf(Foo{}).Name()
+	FooGroupKind        = schema.GroupKind{Group: Group, Kind: FooKind}.String()
+	FooKindAPIVersion   = FooGroupKind + "." + SchemeGroupVersion.String()
+	FooGroupVersionKind = SchemeGroupVersion.WithKind(FooGroupKind)
+)
+
+func init() {
+	SchemeBuilder.Register(&Foo{}, &FooList{})
+}
```

Comment:
we don't need to add such a structure just for test.

Reply:
got it